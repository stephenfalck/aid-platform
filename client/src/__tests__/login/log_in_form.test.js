import React from 'react';
import LogInForm from '../../login/log_in_form';
import { shallow, mount } from 'enzyme';
import renderer, { create } from 'react-test-renderer';
import { TextField } from '@material-ui/core';

describe('<LogInForm>', () => {
it('renders without crashing', () => {
    const wrapper = shallow(<LogInForm />);

    expect(wrapper.find('form')).toBeDefined();
  });

  
  it('renders a correct snapshot', () => {
    let form = create(<LogInForm />)
    expect(form.toJSON()).toMatchSnapshot();
  })


it('Should capture email correctly onChange', () => {
  const component = mount(<LogInForm />);
  const input = component.find('input').at(0);
  input.instance().value = 'hello@email.com';
  input.simulate('change');
  expect(component.state().email).toEqual('hello@email.com');
  component.unmount();
})

/*
it('Should submit the form data to the server', () => {
  const state = {email:'hello@email.com', password:'hello'}
  const component = mount(<LogInForm />);
  component.setState(state)

  const mockSuccessResponse = {};
    const mockJsonPromise = Promise.resolve(mockSuccessResponse); 
    const mockFetchPromise = Promise.resolve({ 
      json: () => mockJsonPromise,
    });
    jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);

  component.find('form').simulate('submit')
  expect(global.fetch).toHaveBeenCalledTimes(1);
  global.fetch.mockClear();
  //component.unmount()
})
*/

/*

it('Should produce errors if form is blank', () => {
  const state = {email: null, password: null}
  const component = mount(<LogInForm />);

  //const mockSuccessResponse = {};
  //  const mockJsonPromise = Promise.resolve(mockSuccessResponse); 
  //  const mockFetchPromise = Promise.resolve({ 
  //    json: () => mockJsonPromise,
  //  });
  //  jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);


  component.setState(state)
  component.find('form').simulate('submit')
  
  expect(component.state().error).toEqual(true)
  component.unmount()
})
  
  
  import React from 'react';import { shallow, mount } from 'enzyme';
import renderer, { create } from 'react-test-renderer';
import Messages from '../../inbox/messages';


describe('<Messages>', () => {
    it('renders without crashing', () => {
        const messages = ['one', 'two', 'three']
        const wrapper = shallow(<Messages messages={messages} />);

        console.log(wrapper.debug());

        expect(wrapper.find('#message-container-item')).toBeDefined();
        expect(wrapper.find('.messsages')).toHaveLength(messages.length);
      });
})
  */
})
