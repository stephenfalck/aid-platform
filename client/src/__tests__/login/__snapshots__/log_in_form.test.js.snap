// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogInForm> renders a correct snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <label
      className="MuiFormLabel-root-16 MuiFormLabel-required-21 MuiInputLabel-required-9 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
      htmlFor="email"
    >
      Email
      <span
        className="MuiFormLabel-asterisk-22"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input-46 MuiInput-input-31"
        disabled={false}
        id="email"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="MuiFormControl-root-1"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <label
      className="MuiFormLabel-root-16 MuiFormLabel-required-21 MuiInputLabel-required-9 MuiInputLabel-root-5 MuiInputLabel-formControl-10 MuiInputLabel-animated-13"
      data-shrink={false}
      htmlFor="adornment-password"
    >
      Password
      <span
        className="MuiFormLabel-asterisk-22"
      >
         *
      </span>
    </label>
    <div
      className="MuiInputBase-root-36 MuiInput-root-23 MuiInput-underline-27 MuiInputBase-formControl-37 MuiInput-formControl-24 MuiInputBase-adornedEnd-41"
      onClick={[Function]}
    >
      <input
        aria-invalid={false}
        className="MuiInputBase-input-46 MuiInput-input-31 MuiInputBase-inputType-49 MuiInput-inputType-34 MuiInputBase-inputAdornedEnd-52"
        disabled={false}
        id="adornment-password"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        required={true}
        type="password"
        value=""
      />
      <div
        className="MuiInputAdornment-root-53 MuiInputAdornment-positionEnd-56"
      >
        <button
          aria-label="Toggle password visibility"
          className="MuiButtonBase-root-64 MuiIconButton-root-58"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-63"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-67"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0zm0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-86"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    className="login-error-message"
    style={
      Object {
        "display": "none",
      }
    }
  />
  <button
    aria-label="Log in"
    className="MuiButtonBase-root-64 MuiFab-root-76 MuiFab-extended-80 MuiFab-secondary-79"
    disabled={false}
    id="login-button"
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    style={
      Object {
        "marginBottom": "10px",
        "marginTop": "15px",
        "width": "100%",
      }
    }
    tabIndex="0"
    type="submit"
  >
    <span
      className="MuiFab-label-77"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root-67"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 2L4.5 20.29l.71.71L12 18l6.79 3 .71-.71z"
        />
      </svg>
      Log in
    </span>
    <span
      className="MuiTouchRipple-root-86"
    />
  </button>
</form>
`;
